File: 2020 kwiecien/python/1.py
with open("dane4.txt", "r") as f:
    liczby = list(map(int, f.read().strip().split('\n')))

maxx, minx = 0, 2e9
for i in range(1, len(liczby)):
    if abs(liczby[i] - liczby[i - 1]) > maxx: maxx = abs(liczby[i] - liczby[i - 1])
    if abs(liczby[i] - liczby[i - 1]) < minx: minx = abs(liczby[i] - liczby[i - 1])

print("Największa:", maxx)
print("Najemniejsza:", minx)

File: 2020 kwiecien/c++/1.cpp
#include <iostream>
#include <fstream>
#include <algorithm>
using namespace std;

int a[1005];

int main()
{
    ifstream in("dane4.txt");

    for (int i = 1; i <= 1000; i++)
        in >> a[i];
    
    int maxx = 0, minx = 2e9;
    for (int i = 2; i <= 1000; i++)
    {
        maxx = max(maxx, abs(a[i] - a[i - 1]));
        minx = min(minx, abs(a[i] - a[i - 1]));
    }

    cout << "Najwieksza: " << maxx << '\n';
    cout << "Najemniejsza: " << minx << '\n';
}